features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/resource-access-control-v2.xml

```diff
@@ -1322,6 +1322,47 @@
         <Scopes>internal_custom_authenticator_delete</Scopes>
     </Resource>
 
+    <!-- User share API -->
+    <Resource context="(.*)/api/server/v1/users/share(.*)" secured="true" http-method="POST">
+        <Scopes>internal_user_share</Scopes>
+    </Resource>
+    <Resource context="(.*)/api/server/v1/users/share-with-all(.*)" secured="true" http-method="POST">
+        <Scopes>internal_user_share</Scopes>
+    </Resource>
+    <Resource context="(.*)/api/server/v1/users/unshare(.*)" secured="true" http-method="POST">
+        <Scopes>internal_user_unshare</Scopes>
+    </Resource>
+    <Resource context="(.*)/api/server/v1/users/unshare-with-all(.*)" secured="true" http-method="POST">
+        <Scopes>internal_user_unshare</Scopes>
+    </Resource>
+    <Resource context="(.*)/api/server/v1/users/(.*)/shared-organizations" secured="true" http-method="GET">
+        <Scopes>internal_user_shared_access_view</Scopes>
+    </Resource>
+    <Resource context="(.*)/api/server/v1/users/(.*)/shared-roles" secured="true" http-method="GET">
+        <Scopes>internal_user_shared_access_view</Scopes>
+    </Resource>
+
+    <!-- [Organization] User share API -->
+    <Resource context="(.*)/o/api/server/v1/users/share(.*)" secured="true" http-method="POST">
+        <Scopes>internal_org_user_share</Scopes>
+    </Resource>
+    <Resource context="(.*)/o/api/server/v1/users/share-with-all(.*)" secured="true" http-method="POST">
```

Comment:
/o/ paths should come prior to the tenant APIs
otherwise org path regex will not pick first when analysing the code